src/hotspot/share/gc/serial/defNewGeneration.cpp

```diff
@@ -172,11 +174,12 @@ class AdjustWeakRootClosure: public OffHeapScanClosure {
 
 class KeepAliveClosure: public OopClosure {
   DefNewGeneration* _young_gen;
-  HeapWord*         _young_gen_end;
+  HeapWord*         _old_gen_boundary;
   CardTableRS* _rs;
 
   bool is_in_young_gen(void* p) const {
-    return p < _young_gen_end;
+    bool is_in_lower_region = p < _old_gen_boundary;
+    return SwapSerialGCGenerations ^ is_in_lower_region;
```

Comment:
(copied from other PR)

This math (and in the other locations) was confusing to me, and I've realized it was because I read _old_gen_boundary as the end of the old gen region, Saint confirmed this is the boundary between old and young regardless of the order in memory.

I'd like to suggest we rename this simply "_gen_boundary" as it is the boundary between the regions and is independent of the order of the regions